src/Middleware/HttpLogging/src/HttpLoggingMiddleware.cs

```diff
@@ -62,41 +59,62 @@ private async Task InvokeInternal(HttpContext context)
         var loggingAttribute = context.GetEndpoint()?.Metadata.GetMetadata<HttpLoggingAttribute>();
         var loggingFields = loggingAttribute?.LoggingFields ?? options.LoggingFields;
 
+        var logContext = new HttpLoggingContext(context)
+        {
+            LoggingFields = loggingFields,
+            RequestBodyLogLimit = options.RequestBodyLogLimit,
+            ResponseBodyLogLimit = options.ResponseBodyLogLimit,
+        };
+
+        if (loggingAttribute?.IsRequestBodyLogLimitSet is true)
+        {
+            logContext.RequestBodyLogLimit = loggingAttribute.RequestBodyLogLimit;
+        }
+        if (loggingAttribute?.IsResponseBodyLogLimitSet is true)
+        {
+            logContext.ResponseBodyLogLimit = loggingAttribute.ResponseBodyLogLimit;
+        }
+
+        for (var i = 0; i < _interceptors.Length; i++)
+        {
+            _interceptors[i].OnRequest(logContext);
+        }
+
+        loggingFields = logContext.LoggingFields;
+
         if ((HttpLoggingFields.Request & loggingFields) != HttpLoggingFields.None)
         {
             var request = context.Request;
-            var list = new List<KeyValuePair<string, object?>>(
-                request.Headers.Count + DefaultRequestFieldsMinusHeaders);
 
             if (loggingFields.HasFlag(HttpLoggingFields.RequestProtocol))
             {
-                AddToList(list, nameof(request.Protocol), request.Protocol);
+                logContext.Add(nameof(request.Protocol), request.Protocol);
```

Comment:
These are all adding, but what if the user set them to a redacted or custom value? You now will print both the users' one and the one added by the middleware.

Reply:
If the user set one of these values themselves then they need to remove it from LoggingFields. Maybe I'll add an Enable/Disable(HttpLoggingFields) API to the context to make that more obvious.